File: Python_Scripting_for_Computational_Science_Third_Edition/py/mixed/Grid2D/C/f2py/gridloop.c
typedef double (*Fxy)(double x, double y);

#define index(a, i, j) a[j*ny + i]

void gridloop2(double *a, double *xcoor, double *ycoor,
	       int nx, int ny, Fxy func1)
{
  int i, j;
  for (i=0; i<nx; i++) {
    for (j=0; j<ny; j++) {
      index(a, i, j) = func1(xcoor[i], ycoor[j]);
    }
  }
}

File: Python_Scripting_for_Computational_Science_Third_Edition/py/mixed/Grid2D/Matlab/gridloop.m
n = 2000;
a = zeros([n,n]);
xcoor=0:(1/n):1;
ycoor=0:(1/n):1;
for i = 1:n,
   for j = 1:n,
      a(i,j) = f(xcoor(i), xcoor(j));
   end
end
cputime

